contracts/platform/registry/Registry.sol

```diff
@@ -70,16 +70,22 @@ contract Registry is Destructible, RegistryInterface, Upgradable {
         require(endpoint != bytes32(0), "Error: Can't initiate null endpoint");
 
         setCurve(msg.sender, endpoint, curve);
+
         db.pushBytesArray(keccak256(abi.encodePacked('oracles', msg.sender, 'endpoints')), endpoint);
-        db.setBytes32(keccak256(abi.encodePacked('oracles', msg.sender, endpoint, 'broker')), bytes32(broker));
+
+       
+        db.setBytes32(keccak256(abi.encodePacked('oracles', msg.sender, endpoint, 'broker')), bytes32(uint256(broker) << 96));
 
         emit NewCurve(msg.sender, endpoint, curve, broker);
 
         return true;
     }
 
+    function toBytes(address a) public pure returns (bytes memory) {
```

Comment:
this can be removed. 

originally was going to add this function to convert address to bytes1 but it was not necessary.